algonaut_transaction/src/api_model.rs

```diff
@@ -134,7 +138,22 @@ impl From<Transaction> for ApiTransaction {
                 api_t.local_state_schema =
                     call.to_owned().local_state_schema.and_then(|s| s.into());
                 api_t.extra_pages = num_as_api_option(call.extra_pages);
-            }
+            },
+            TransactionType::StateProofTransaction(stpf) => {
+                api_t.state_proof_type = {
+                    let stpf = stpf.state_proof_type;
+                    match stpf {
+                        StateProofType::StateProofBasic => Some(0),
```

Comment:
I assume that if the field is 0 it doesn't have to be sent to the API, so this should be set to `None`. We decided to be very explicit about these conversions, with functions like `num_as_api_option`. 

For some context, in Rust the "skip 0 values" policy is unpopular enough that it's not possible to set it universally, it can be done per field via macros, but again for clarity, we decided to consistently skip only `None` and do manual conversions to map to `None` where needed.

Reply:
state proof txns are not sent via the API but internal in the node so I'm not sure they'll ever need to be serialized and sent in, does that make this match case inappropriate to add?

Reply:
How is it handled in the other SDKs? The transactions could be serialized to be persisted or sent to a third party (which would also go through the conversion to `ApiTransaction`) and fields should not disappear. That said, given the current state of the SDK and that this is likely not a very common use case, I'd be fine with skipping it for now.